src/User.php

```diff
@@ -3492,6 +3494,28 @@ public static function processMassiveActionsForOneItemtype(
                     $ma->addMessage($item->getErrorMessage(ERROR_RIGHT));
                 }
                 return;
+
+            case 'delete_emails':
+                // Check rights
+                if (!self::canUpdate()) {
+                    $ma->itemDone($item->getType(), $ids, MassiveAction::ACTION_NORIGHT);
+                    $ma->addMessage($item->getErrorMessage(ERROR_RIGHT));
+                    return;
+                }
+
+                foreach ($ids as $id) {
```

Comment:
Rights checks should also be done on each item.
```suggestion
                foreach ($ids as $id) {
                    if (!$item->can($id, UPDATE)) {
                        $ma->itemDone($item->getType(), $id, MassiveAction::ACTION_NORIGHT);
                        $ma->addMessage($item->getErrorMessage(ERROR_RIGHT));
                    }
```